airbyte-server/src/test/java/io/airbyte/server/handlers/DestinationHandlerTest.java

```diff
@@ -182,35 +182,40 @@ void testDeleteDestination() throws JsonValidationException, ConfigNotFoundExcep
 
   @Test
   void testUpdateDestination() throws JsonValidationException, ConfigNotFoundException, IOException {
+    final String updatedDestName = "my updated dest name";
     final JsonNode newConfiguration = destinationConnection.getConfiguration();
-
     ((ObjectNode) newConfiguration).put("apiKey", "987-xyz");
 
     final DestinationConnection expectedDestinationConnection = Jsons.clone(destinationConnection)
+        .withName(updatedDestName)
         .withConfiguration(newConfiguration)
         .withTombstone(false);
+
     final DestinationUpdate destinationUpdate = new DestinationUpdate()
+        .name(updatedDestName)
         .destinationId(destinationConnection.getDestinationId())
-        .name(destinationConnection.getName())
         .connectionConfiguration(newConfiguration);
 
-    when(configRepository.getDestinationConnection(destinationConnection.getDestinationId()))
-        .thenReturn(destinationConnection)
-        .thenReturn(expectedDestinationConnection);
-    when(specFetcher.execute(imageName)).thenReturn(connectorSpecification);
-    when(configRepository.getStandardDestinationDefinition(standardDestinationDefinition.getDestinationDefinitionId()))
-        .thenReturn(standardDestinationDefinition);
     when(secretsProcessor
         .copySecrets(destinationConnection.getConfiguration(), newConfiguration, destinationDefinitionSpecificationRead.getConnectionSpecification()))
             .thenReturn(newConfiguration);
     when(secretsProcessor.maskSecrets(newConfiguration, destinationDefinitionSpecificationRead.getConnectionSpecification()))
         .thenReturn(newConfiguration);
-    when(configurationUpdate.destination(destinationConnection.getDestinationId(), newConfiguration)).thenReturn(expectedDestinationConnection);
+    when(configRepository.getStandardDestinationDefinition(standardDestinationDefinition.getDestinationDefinitionId()))
+        .thenReturn(standardDestinationDefinition);
+    when(configRepository.getDestinationDefinitionFromDestination(destinationConnection.getDestinationId()))
+        .thenReturn(standardDestinationDefinition);
+    when(configRepository.getDestinationConnection(destinationConnection.getDestinationId()))
+        .thenReturn(destinationConnection)
+        .thenReturn(expectedDestinationConnection);
```

Comment:
```suggestion
    when(configRepository.getDestinationConnection(destinationConnection.getDestinationId()))       
        .thenReturn(expectedDestinationConnection);
```

This should fix your test that is failing...

The reason is that the `sourceHandler` and `destinationHandler` are not doing the exact same thing when calling `getSpec` and `getSpecFromSourceId`:
![Screenshot 2021-06-07 at 16 01 05](https://user-images.githubusercontent.com/7718175/121030618-ee849980-c7a9-11eb-9ba1-10e964c9a14d.png)
![Screenshot 2021-06-07 at 16 01 27](https://user-images.githubusercontent.com/7718175/121030821-1e33a180-c7aa-11eb-8e63-3dc05642afb0.png)

So your mocks are not called twice in the destination but twice in the source case, thus my suggestion:
![Screenshot 2021-06-07 at 16 02 09](https://user-images.githubusercontent.com/7718175/121030927-35728f00-c7aa-11eb-8977-8ac7b07a30a1.png)